packages/api-client/src/connections/AuthApi.ts

```diff
@@ -52,3 +53,5 @@ export class AuthApi extends BaseApi {
     return { accessToken, refreshToken, accessPolicy, email, user };
   }
 }
+
+export interface AuthApiInterface extends PublicInterface<AuthApi> {}
```

Comment:
So I appreciate what I'm doing here is a little bit unfashionable. 
Typically in this scenario you:
1. Define an interface with the public methods you want it to have
2. Define a concrete implementation of the interface
3. Define a mock implementation of the interface

The downside to that pattern is that you have to remember to update the interface whenever you want to add methods. It's also a bit more boilerplate-y.

Instead what we're doing here is inferring an interface from the public methods of the concreate implementation, which means we can just freely add methods to the concrete implementation and the interface will automatically update. But if this feels too awkward then I'm happy to switch to the more standard way of doing things 👍

Reply:
No this is great, I like it!